finetune/lora.py

```diff
@@ -193,6 +194,10 @@ def train(
 
         input_ids, targets = get_batch(fabric, train_data, longest_seq_ix if iter_num == 1 else None)
 
+        # Truncate if needed
```

Comment:
Should truncation be done to validation too? From a memory perspective is unnecessary, but from a modelling performance, I don't think you would want to measure the model on longer sequences than what the model has trained with.

This would get done automatically if you move this to `get_batch`